project/TessellationCiRelease.scala

```diff
@@ -0,0 +1,164 @@
+      case None =>
+        import scala.sys.process._
+        val identifier = """([^\/]+?)"""
+        val GitHubHttps =
+          s"https://github.com/$identifier/$identifier(?:\\.git)?".r
+        val GitHubGit = s"git://github.com:$identifier/$identifier(?:\\.git)?".r
+        val GitHubSsh = s"git@github.com:$identifier/$identifier(?:\\.git)?".r
+        try {
+          val remote = List("git", "ls-remote", "--get-url", "origin").!!.trim()
+          remote match {
+            case GitHubHttps(user, repo) => Some(gitHubScmInfo(user, repo))
+            case GitHubGit(user, repo)   => Some(gitHubScmInfo(user, repo))
+            case GitHubSsh(user, repo)   => Some(gitHubScmInfo(user, repo))
+            case _                       => None
+          }
+        } catch {
+          case NonFatal(_) => None
+        }
+    }
+  )
+
+  override lazy val globalSettings: Seq[Def.Setting[_]] = List(
+    (Test / publishArtifact) := false,
+    commands += Command.command("tessellation-ci-release") { currentState =>
+      val version = getVersion(currentState)
+      val isSnapshot = isSnapshotVersion(version)
+      if (!isSecure) {
+        println("No access to secret variables, doing nothing")
+        currentState
+      } else {
+        println(
+          s"Running ci-release.\n" +
+            s"  branch=$currentBranch"
+        )
+        setupGpg()
+        // https://github.com/olafurpg/sbt-ci-release/issues/64
+        val reloadKeyFiles = "; set pgpSecretRing := pgpSecretRing.value; set pgpPublicRing := pgpPublicRing.value"
+        val publishCommand = "+publishSigned"
+
+        if (isSnapshot) {
+          println(s"Sonatype Central does not accept snapshots, only official releases. Aborting release.")
+          currentState
+        } else if (!isTag) {
+          println(s"No tag published. Cannot publish an official release without a tag and Sonatype Central does not accept snapshot releases. Aborting release.")
+          currentState
+        } else {
+          println("Tag push detected, publishing a stable release")
+          reloadKeyFiles ::
+            sys.env.getOrElse("CI_CLEAN", "; clean ; sonatypeBundleClean") ::
+            publishCommand ::
+            sys.env.getOrElse("CI_SONATYPE_RELEASE", "sonatypeCentralRelease") ::
+            currentState
+        }
+      }
+    }
+  )
+
+  override lazy val projectSettings: Seq[Def.Setting[_]] = List(
+    version := (ThisBuild / version).value,
+    publishConfiguration := publishConfiguration.value.withOverwrite(true),
+    publishLocalConfiguration := publishLocalConfiguration.value.withOverwrite(true),
+    publishTo := sonatypePublishToBundle.value
+  )
+
+  private def getVersion(state: State): String =
+    (ThisBuild / version).get(Project.extract(state).structure.data) match {
+      case Some(v) => v
+      case None    => throw new NoSuchFieldError("version")
+    }
+
+  private def isSnapshotVersion(v: String): Boolean = v.endsWith("-SNAPSHOT")
```

Comment:
The version may end with `-SNAPSHOT-dirty` and it should be also considered a snapshot version 🤔 instead of `endsWith` I'd use `contains("-SNAPSHOT")`